cmip6_downscaling/methods/regions.py

```diff
@@ -0,0 +1,99 @@
+from typing import Any, Dict, Tuple, Union
+
+import numpy as np
+import regionmask
+import xarray as xr
+
+
+def generate_subdomains(
+    ex_output_grid: Union[xr.Dataset, xr.DataArray],
+    buffer_size: Union[float, int],
+    region_def: str = 'ar6',
+) -> Tuple[Dict[Union[int, float], Any], xr.DataArray]:
+    """
+    Given an example output grid, determine all subdomains that need to be process in order to generate the final output.
+    Outputs the list of bounding boxes for each subdomain considering the buffer size, as well as a mask in the resolution of the example output grid specifying
+    which subdomain's value to use for each grid cell.
+
+    Parameters
+    ----------
+    ex_output_grid : xarray dataarray or dataset. example output grid definition. both the bounding box and resolution in lat/lon directions will be used.
+    buffer_size    : int/float in unit of degree. for each subdomain, how much buffer area to run
+    region_def     : str. whether to use SREX region definition or ar6 region definition to define subdomain.
+                     see the docs https://regionmask.readthedocs.io/en/stable/defined_scientific.html for more details.
```

Comment:
Suggest reformatting to align with numpy-doc style. Something like this:

```suggestion
    ex_output_grid : xarray.DataArray or xarray.Dataset
        Example output grid definition. both the bounding box and resolution in lat/lon directions will be used.
    buffer_size : int or float
        Buffer size in unit of degree. for each subdomain, how much buffer area to run
    region_def : str
        Subregion definition name. Options are `'ar6'` or `'srex'`. See the docs https://regionmask.readthedocs.io/en/stable/defined_scientific.html for more details.
```

Reply:
It will probably be faster for both of us if I just say -- let's format all doc strings with the numpy-doc style.